src/praisonai/praisonai/bots/_presentation_approval.py

```
@@ -0,0 +1,267 @@
+        self._approval_futures[approval_id] = future
+        
+        # Create approval presentation
+        presentation = self._create_approval_presentation(
+            approval_id, tool_name, arguments, risk_level
+        )
+        
+        # Send presentation if channel function provided
+        if channel_send_func and target:
+            try:
+                await channel_send_func(target, presentation)
+            except Exception as e:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)